Gems/Atom/Utils/Code/Include/Atom/Utils/ImGuiGpuProfiler.inl

```diff
@@ -1024,6 +1117,248 @@ namespace AZ
+                break;
+            case (1): // Sort by buffer/image name
+                AZStd::sort(m_tableRows.begin(), m_tableRows.end(),
+                    [ascending, &getName](const TableRow& lhs, const TableRow& rhs){
+                        const auto lhsName = AZStd::visit(getName, lhs.m_variant).GetStringView();
+                        const auto rhsName = AZStd::visit(getName, rhs.m_variant).GetStringView();
+                        return ascending ? lhsName < rhsName : lhsName > rhsName;
+                    });
+                break;
+            case (2): // Sort by memory usage
+                AZStd::sort(m_tableRows.begin(), m_tableRows.end(),
+                    [ascending, &getSize](const TableRow& lhs, const TableRow& rhs){
+                        const float lhsSize = AZStd::visit(getSize, lhs.m_variant);
+                        const float rhsSize = AZStd::visit(getSize, rhs.m_variant);
+                        return ascending ? lhsSize < rhsSize : lhsSize > rhsSize;
+                    });
+                break;
+            }
+            sortSpecs->SpecsDirty = false;
+        }
+
+        inline void ImGuiGpuMemoryView::DrawTable()
+        {
+            if (ImGui::BeginTable("Table", 4, ImGuiTableFlags_Borders | ImGuiTableFlags_Sortable | ImGuiTableFlags_Resizable))
+            {
+                ImGui::TableSetupColumn("Parent pool");
+                ImGui::TableSetupColumn("Name");
+                ImGui::TableSetupColumn("Size (MB)", 0, 100.0f);
+                ImGui::TableSetupColumn("BindFlags", ImGuiTableColumnFlags_NoSort);
+                ImGui::TableHeadersRow();
+                ImGui::TableNextColumn();
+
+                ImGuiTableSortSpecs* sortSpecs = ImGui::TableGetSortSpecs();
+                if (sortSpecs && sortSpecs->SpecsDirty){
+                    SortTable(sortSpecs);
+                }
+
+                // Draw each row in the table
+                for (const auto& tableRow : m_tableRows) {
+                    AZStd::string rowName = "";
+                    float size = 0.0;
+                    AZStd::string bindStrings = ""; 
+
+                    // Get the variant's fields
+                    AZStd::visit(
+                        [&rowName, &size](auto&& val)
+                        {
+                            rowName = val->m_name.GetCStr();
+                            size = 1.0f * val->m_sizeInBytes / GpuProfilerImGuiHelper::MB;
+                        }, tableRow.m_variant);
+                    if (auto buf = AZStd::get_if<Buffer*>(&tableRow.m_variant))
+                    {
+                        bindStrings = GpuProfilerImGuiHelper::GetBufferBindStrings((*buf)->m_bindFlags);
+                    }
+                    else if (auto img = AZStd::get_if<Image*>(&tableRow.m_variant))
+                    {
+                        bindStrings = GpuProfilerImGuiHelper::GetImageBindStrings((*img)->m_bindFlags);
+                    }
```

Comment:
Maybe using a variant to store the Image or Buffer is not the right approach here, this is pretty ugly.